applications/StructuralMechanicsApplication/custom_elements/base_shell_element.h

```diff
@@ -0,0 +1,334 @@
+    void EquationIdVector(EquationIdVectorType& rResult, ProcessInfo& CurrentProcessInfo) override;
+
+    /**
+    * determines the elemental list of DOFs
+    * @param ElementalDofList: the list of DOFs
+    * @param rCurrentProcessInfo: the current process info instance
+    */
+    void GetDofList(DofsVectorType& rElementalDofList, ProcessInfo& rCurrentProcessInfo) override;
+
+
+    void GetValuesVector(Vector& rValues, int Step = 0) override;
+
+    void GetFirstDerivativesVector(Vector& rValues, int Step = 0) override;
+
+    void GetSecondDerivativesVector(Vector& rValues, int Step = 0) override;
+
+    void ResetConstitutiveLaw() override;
+
+    void CalculateMassMatrix(MatrixType& rMassMatrix, ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateDampingMatrix(MatrixType& rDampingMatrix, ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                              VectorType& rRightHandSideVector,
+                              ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateRightHandSide(VectorType& rRightHandSideVector,
+	                            ProcessInfo& rCurrentProcessInfo) override;
+
+    /**
+    * This method provides the place to perform checks on the completeness of the input
+    * and the compatibility with the problem options as well as the contitutive laws selected
+    * It is designed to be called only once (or anyway, not often) typically at the beginning
+    * of the calculations, so to verify that nothing is missing from the input
+    * or that no common error is found.
+    * @param rCurrentProcessInfo
+    * this method is: MANDATORY
+    */
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+    /**
+    * returns the used integration method. In the general case this is the
+    * default integration method of the used geometry. I an other integration
+    * method is used the method has to be overwritten within the element
+    * @return default integration method of the used Geometry
+    */
+    IntegrationMethod GetIntegrationMethod() const override
+    {
+        return mIntegrationMethod;
+    }
+    ///@}
+    ///@name Access
+    ///@{
+
+    void SetCrossSectionsOnIntegrationPoints(std::vector< ShellCrossSection::Pointer >& crossSections);
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const;
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const;
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream& rOStream) const;
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    IntegrationMethod mIntegrationMethod = GeometryData::GI_GAUSS_2;
```

Comment:
mmm
do you need to store this? it is one int per element, which i think is largely unneeded